### ISSUE TITLE
[BUG] ConvertTo-PnPPage Access Denied Error

### ISSUE THREAD

**Issue by PrasadMadhavan** [pnp/pnpframework#178]
**### Reporting an Issue**
I am using Version1.3.0 PnP.Powershell. to convert a classic page to modern page on another site collection.  But it fails when I try to execute below command

ConvertTo-PnPPage -Identity "sourcepage.aspx" -PublishingPage -TargetWebUrl "https://xxxxx.sharepoint.com/sites/modernuat/EmployeeResources"

### Expected behavior 
Classic Page should get copied over to the sitepages library under the new modern communication site. 

### Actual behavior 
I get an error message like below. It says I do not have permissions.   I have site collection admin access on both Source and destination site collections. Multifactor authentication is enabled and I have used -useWebLogin for authentication. I don't have permission to access SP Tenant Admin pages. 

ConvertTo-PnPPage : {"odata.error":{"code":"-2147024891,
System.UnauthorizedAccessException","message":{"lang":"en-US","value":"Access denied. You do not have permission to
perform this action or access this resource."}}}
ServerErrorTraceCorrelationId: 7b43aa9f-9027-b000-bfd5-6bcf98bf8299
At line:1 char:1

ConvertTo-PnPPage -Identity "sourcepage.aspx" -PublishingPa ...
  + CategoryInfo          : WriteError: (:) [ConvertTo-PnPPage], Exception
  + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Pages.ConvertToClientSidePage

### Steps to reproduce behavior

 Connect-PnPOnline $sourceSiteUrl -UseWebLogin
ConvertTo-PnPPage -Identity "sourcepage.aspx" -PublishingPage -TargetWebUrl "https://xxxxx.sharepoint.com/sites/modernuat/EmployeeResources"

### What is the version of the Cmdlet module you are running?
PnP.Powershell 1.3.0

### Which operating system/environment are you running PnP PowerShell on?
- [ ] Windows

**Comment by LogicElite** [comment]
> Can you try to repeat the error for occuring and immediately after that enter:
> 
> ```powershell
> Get-PnPException
> ```
> 
> Please post the output of that cmdlet here. It could help us to determine exactly where it goes wrong on your end.

Hi @erwinvanhunen,

First of all, thanks for your help!

I'm also experiencing this issue. Please find below the exception returned by PnP

Get-PnPException : Unable to cast object of type 'System.Management.Automation.IncompleteParseException' to type 'System.Management.Automation.ErrorRecord'.
At line:1 char:1
+ Get-PnPException
+ ~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-PnPException], InvalidCastException
    + FullyQualifiedErrorId : System.InvalidCastException,PnP.PowerShell.Commands.Base.GetException

**Comment by jansenbe** [comment]
@LogicElite / @PrasadMadhavan : would it be possible to create a Fiddler trace and share that? I've tried to repro within my lab but did not yet succeed, a Fiddler trace will show which specific API call fails and why that happens. You can share the Fiddler with bjansen@microsoft.com in case you want to keep it private. Thx.

**Comment by PrasadMadhavan** [comment]
I have emailed you the fiddler trace, Thanks.

**Comment by jansenbe** [comment]
Thanks for sharing the Fiddler trace @PrasadMadhavan. I see where things fail, but for now I've no idea why it happens. Can you always reproduce this issue or is it more randomly?

Can you  try to to connect using -UseWebLogin and then call  `Add-PnPPage -Name test.aspx`: this will create a modern page using the same API's as page transformation is doing, want to see if this shows the same issue or not.

**Comment by jansenbe** [comment]
@LogicElite / @PrasadMadhavan : can you also let us know what PS version you're using? 

![image](https://user-images.githubusercontent.com/7451219/108185629-cc88dd80-710c-11eb-8b1f-ee392c00af5d.png)

**Comment by jansenbe** [comment]
@LogicElite / @PrasadMadhavan : we managed to reproduce this with using PnP PowerShell in PowerShell 5 and have fixed this. Please retest with tomorrow's nightly build or alternatively install PowerShell 7 and use that.

**Comment by PrasadMadhavan** [comment]
I just upgraded to PowerShell and the error has gone! Thanks a lot for your help.

**Comment by LogicElite** [comment]
> @LogicElite / @PrasadMadhavan : we managed to reproduce this with using PnP PowerShell in PowerShell 5 and have fixed this. Please retest with tomorrow's nightly build or alternatively install PowerShell 7 and use that.

Hi @jansenbe,

I tried using PowerShell 7, it create the page and the target but is unfortunately throwing a critical error for some pages stored within the Enterprise Wiki. And for some reason, the page created now have + character instead of space. Most of the pages contains special characters such as : é, à, etc.. I was not experiencing these issues 1 month ago.

Please find the critical error received below: 

Error: [Summary] A critical error occurred - transformation did not complete Error: SharePoint Rest service exception StackTrace:    at PnP.Core.Services.BatchClient.ProcessSharePointRestBatchResponseContent(Batch batch, String batchResponse)
   at PnP.Core.Services.BatchClient.ProcessSharePointRestBatchResponse(SPORestBatch restBatch, String batchResponse)
   at PnP.Core.Services.BatchClient.ExecuteSharePointRestBatchAsync(Batch batch)
   at PnP.Core.Services.BatchClient.ExecuteBatch(Batch batch)
   at PnP.Core.Model.BaseDataModel`1.RequestAsync(ApiCall apiCall, HttpMethod method, String operationName)
   at PnP.Core.Model.SharePoint.FileCollection.AddTemplateFileAsync(String serverRelativePageName, TemplateFileType templateFileType)
   at PnP.Core.Model.SharePoint.Page.SaveAsync(String pageName)
   at PnP.Core.Model.SharePoint.Page.Save(String pageName)
   at PnP.Framework.Modernization.Publishing.PublishingPageTransformator.Transform(PublishingPageTransformationInformation publishingPageTransformationInformation)

Thanks for your help and support.

**Comment by HenryatSolu** [comment]
Hi @jansenbe

I am trying to modernize publishing pages directly from SP2013 to SPO and am testing it with: 
`ConvertTo-PnPPage -Identity "test.aspx" -PublishingPage -TargetConnection $target -Connection $source`

Both connections seem alive. However I keep getting error 403. This also happens when I am execute Get-PnP.ListItem or any other Get at the source. I upgraded to PowerShell 7.2.5 but that did not resolve the issue in my case. 

This is what I get with `Get-PnPExeption`
`Stacktrace       :    at Microsoft.SharePoint.Client.ClientContextExtensions.GetOnPremisesRequestDigestInfoAsync(ClientContext context)
                      at Microsoft.SharePoint.Client.ClientContextExtensions.GetOnPremisesRequestDigestAsync(ClientContext context)
                      at PnP.Framework.AuthenticationManager.<>c.<ConfigureOnPremisesContext>b__69_0(Object sender, WebRequestEventArgs webRequestEventArgs)
                      at Microsoft.SharePoint.Client.ClientRuntimeContext.OnExecutingWebRequest(WebRequestEventArgs args)
                      at Microsoft.SharePoint.Client.ClientRuntimeContext.FireExecutingWebRequestEvent(WebRequestEventArgs args)
                      at Microsoft.SharePoint.Client.ClientRequest.SetupServerQuery(ChunkStringBuilder sb)
                      at Microsoft.SharePoint.Client.ClientRequest.ExecuteQueryToServerAsync(ChunkStringBuilder sb)
                      at Microsoft.SharePoint.Client.ClientRequest.ExecuteQueryAsync()
                      at Microsoft.SharePoint.Client.ClientRuntimeContext.ExecuteQueryAsync()
                      at Microsoft.SharePoint.Client.ClientContext.ExecuteQueryAsync()
                      at Microsoft.SharePoint.Client.ClientContextExtensions.ExecuteQueryImplementation(ClientRuntimeContext clientContext, Int32 retryCount, String userAgent)
                      at Microsoft.SharePoint.Client.ClientObjectExtensions.EnsurePropertyImplementation[T,TResult](T clientObject, Expression`1 propertySelector)
                      at Microsoft.SharePoint.Client.ClientObjectExtensions.EnsureProperty[T,TResult](T clientObject, Expression`1 propertySelector)
                      at PnP.Framework.Modernization.Cache.CacheManager.GetPublishingPagesLibraryName(ClientContext context)
                      at PnP.PowerShell.Commands.Pages.ConvertToClientSidePage.ExecuteCmdlet() in c:\build\src\Commands\Pages\ConvertToPage.cs:line 176
                      at PnP.PowerShell.Commands.PnPSharePointCmdlet.ProcessRecord() in c:\build\src\Commands\Base\PnPSharePointCmdlet.cs:line 114
ScriptLineNumber : 1`

Thank you  in advance for your help.

**Comment by HenryatSolu** [comment]
> Both connections seem alive. However I keep getting error 401. This also happens when I am execute Get-PnP.ListItem or any other Get at the source. I upgraded to PowerShell 7.2.5 but that did not resolve the issue in my case.

Update to my case:  Try to connect without the switch -TransformationOnPrem revealed this:

`Connect-PnPOnline: There was an error parsing WS-Trust response from the endpoint. This may occur if there is an issue with your ADFS configuration. See https://aka.ms/msal-net-iwa-troubleshooting for more details. Error Message: Federated service at https://autologon.microsoftazuread-sso.com/xxxyyyzzz.com/winauth/trust/2005/usernamemixed?client-request-id=e3d183f7-a6c8-4dcd-aa5a-35afd57ff079 returned error: Authentication Failure.`

So it seems it boils down to ADFS and what I am looking at now is how to go around it. And just to be clarify: first I got an 403 error message using PowerShell 5.x. Using Power what I got was 401. This is why I posted in this thread in the first place.
